Add support of native type mappings from Prisma schema

pasha-vuiko opened the pull request.

pasha-vuiko committed:
Add support of native PostgreSQL time type

pasha-vuiko committed:
Fix possible issue with native type

pasha-vuiko committed:
Add time support in MySQL

pasha-vuiko committed:
Format code

pasha-vuiko commented:
@Sukairo-02 Could you please review?🙂

AlexanderArvidsson commented:
Glad to see this and other repos getting some use out of the example I posted in the original Prisma issue :D

Keep in mind that this approach will traverse the AST tree from scratch on every call, so exponential complexity. If you want to optimize it, you can traverse it once and make a lookup map ;)
I haven't run into any significant performance problems due to this in my generator, but something to remember.

Sukairo-02 commented:
Yeah, sorry for not responding for a long time, I'll look into it the moment I get the chance to, really busy right now.

pasha-vuiko committed:
chore: update Prisma to v6.0.0 in order to use already exposed native types attributes also update some other dependencies to not breaking versions

pasha-vuiko commented:
Hey, hey, I contributed to Prisma to expose native type attributes, now with Prisma v6.0.0 it is avaliable (though it is not mentioned in patch notes) so I updated the Prisma version in the package and updated the code.

No hurry on merging the PR, I understand, that currently Drizzle Team is actively working on v1.0.0 release 😉

Maybe I'll add support for more native types in coming days if I have time.

pasha-vuiko committed:
feat: add Drizzle mappings for Prisma Date, Time, Timetz, Timestamp, Timestamptz, Json, JsonB native mappings for PostgreSQL

pasha-vuiko committed:
fix: fix uuid attribute mapping

pasha-vuiko committed:
feat: add support for native UUID pg type

pasha-vuiko commented:
> (Note from the peanut gallery: we are struggling with @db.Date not coming through correctly, which is currently blocking our migration to Drizzle since all such values come through as `Invalid Date`, so would be amazing if this could be merged & released! Not sure what the blocker is 😅)

As I understand all Drizzle development resources are on the v1.0.0 release. So, till it is released I guess my PR will not be merged. As a workaround you can use `patch-package` with the changes from the PR
